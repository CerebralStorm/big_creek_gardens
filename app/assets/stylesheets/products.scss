.product-tile {
  height: 360px;
}

.product-title {
  height: 30px;
}

.product-description {
  height: 60px;
  overflow: hidden;
  text-overflow: ellipsis;
}

.product-price {
  height: 20px;
}

.product-cart-add {
  height: 20px;
}

.checkout-container {
  margin-top: 20px;
  margin-bottom: 20px;
}

.checkout-item-small {
  padding: 0px;
}

.checkout-title {
  color: white;
}

.checkout-cart-header {
  background-color: #343a40 !important
}

.checkout-name {
  position: relative;
  white-space: nowrap;
  text-overflow: ellipsis;
  display: block;
  overflow: hidden
}

small.checkout-description {
  position: relative;
  white-space: nowrap;
  text-overflow: ellipsis;
  display: block;
  overflow: hidden
}
.checkout-total {
  border-top: 2px solid black;
}