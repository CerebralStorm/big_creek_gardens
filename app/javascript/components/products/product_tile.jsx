import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { updateCart, loadCart } from '../../actions/cart'

class ProductTile extends React.Component {
  constructor(props) {
    super(props);
    this.addToCart = this.addToCart.bind(this)
  }

  addToCart() {
    let newCart = this.props.cart
    if(newCart[this.props.product.id]) {
      newCart[this.props.product.id]['quantity'] += 1;
    } else {
      newCart[this.props.product.id] = {}
      newCart[this.props.product.id]['quantity'] = 1;
      newCart[this.props.product.id]['product'] = this.props.product;
    }
    console.log('New Cart', newCart);
    this.props.dispatch(updateCart(this.props.currentUser, newCart))
  }

  render() {
    return (
      <div className="product-tile col">
        <img src={this.props.product.image_url}  height="160" width="160" alt="Product Image" className='img-fluid' />
        <div className="product-details">
          <h6>{this.props.product.name}</h6>
          <p className="product-tile-description">{this.props.product.description}</p>
          <p>{this.props.product.price}</p>
          <a href='#' onClick={this.addToCart}><FontAwesomeIcon icon='plus'/>Add to Cart</a>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    currentUser: state.currentUser,
    cart: state.cart
  }
}

export default connect(mapStateToProps)(ProductTile)

