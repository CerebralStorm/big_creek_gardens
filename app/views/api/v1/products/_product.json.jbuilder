json.name product.name
json.description product.description
json.price product.price
json.image product.image