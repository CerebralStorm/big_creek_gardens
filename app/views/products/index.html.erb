<div class="container">
  <div class="section">
    <h1 class="text-center">Products</h1>
    <hr />
    <div class="row">
      <% @products.each do |product| %>
        <div class="product-tile col">
          <%= image_tag product.image.url, height: 160, width: 160, alt: "Product Image", class: 'img-fluid' %>
          <div class="product-details">
            <h6><%= product.name %></h6>
            <p class="product-tile-description"><%= product.description %></p>
            <p><%= product.price %></p>
            <a href='#'><%= fa_icon('plus') %>Add to Cart</a>
          </div>
        </div>
      <% end %>
    </div>
    <div class="clearfix"></div>
  </div>
</div>