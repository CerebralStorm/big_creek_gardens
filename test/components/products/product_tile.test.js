import React from 'react'
import { ProductTile } from '../../../app/javascript/components/products/product_tile'

let product = { id: 1, name: 'test product' }

test('render a ProductTile', () => {
  const wrapper = shallow(<ProductTile product={product} />);
  expect(wrapper).toMatchSnapshot();
});