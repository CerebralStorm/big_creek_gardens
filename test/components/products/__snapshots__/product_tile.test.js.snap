// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a ProductTile 1`] = `
<div
  className="product-tile col-xs-12 col-sm-12 col-md-4 col-lg-3"
>
  <img
    alt="Product Image"
    height="160"
  />
  <div
    className="product-details"
  >
    <h6
      className="product-title p-2"
    >
      test product
    </h6>
    <p
      className="product-description"
    />
    <p
      className="product-price"
    >
      <strong>
        $
      </strong>
    </p>
    <p
      className="product-cart-add"
    >
      <a
        className="btn btn-success"
        href="#"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="mr-1"
          fixedWidth={false}
          flip={null}
          icon="plus"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        Add to Cart
      </a>
    </p>
  </div>
</div>
`;
